sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -123,9 +123,10 @@ static inline _nib_onl_entry_t *_cache_out_onl_entry(const ipv6_addr_t *addr,
             res = tmp;
             res->mode = _EMPTY;
             _override_node(addr, iface, res);
-            /* cstate masked above already */
-            res->info = cstate;
+            /* cstate masked in _nib_nc_add() already */
+            res->info |= cstate;
             res->mode = _NC;
+            clist_rpush(&_next_removable, (clist_node_t *)res);
```

Comment:
in both cases
* in the `if` block
* outside of the `if` block
`clist_rpush(&_next_removable, (clist_node_t *)tmp);` (with the assumption `res == tmp`) is called. Can't you merge both lines and move it above of the `if`?